templates/todo/projects/nodejs-mongo/.repo/terraform/infra/main.tf

```diff
@@ -24,11 +25,110 @@ resource "azurerm_resource_group" "rg" {
+# Deploy key vault
+# ------------------------------------------------------------------------------------------------------
+module "keyvault" {
+  source                   = "../../../../../../common/infra/terraform/core/keyvault"
+  location                 = var.location
+  principal_id             = var.principal_id
+  rg_name                  = azurerm_resource_group.rg.name
+  tags                     = azurerm_resource_group.rg.tags
+  resource_token           = local.resource_token
+  access_policy_object_ids = [module.appserviceapi.IDENTITY_PRINCIPAL_ID]
+  secrets = [
+    {
+      name  = local.cosmos_connection_string_key
+      value = module.cosmos.AZURE_COSMOS_CONNECTION_STRING
+    }
+  ]
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy cosmos
+# ------------------------------------------------------------------------------------------------------
+module "cosmos" {
+  source         = "../../../../../../common/infra/terraform/core/database/cosmos"
+  location       = var.location
+  rg_name        = azurerm_resource_group.rg.name
+  tags           = azurerm_resource_group.rg.tags
+  resource_token = local.resource_token
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy app service plan
+# ------------------------------------------------------------------------------------------------------
+module "appserviceplan" {
+  source         = "../../../../../../common/infra/terraform/core/host/appserviceplan"
+  location       = var.location
+  rg_name        = azurerm_resource_group.rg.name
+  tags           = azurerm_resource_group.rg.tags
+  resource_token = local.resource_token
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy app service web app
+# ------------------------------------------------------------------------------------------------------
+module "appserviceweb" {
+  source         = "../../../../../../common/infra/terraform/core/host/appservice/appservicenode"
+  location       = var.location
+  rg_name        = azurerm_resource_group.rg.name
+  resource_token = local.resource_token
+
+  tags               = merge(local.tags, { azd-service-name : "web" })
+  service_name       = "web"
+  appservice_plan_id = module.appserviceplan.APPSERVICE_PLAN_ID
+  app_settings = {
+    "SCM_DO_BUILD_DURING_DEPLOYMENT"        = "false"
+    "APPLICATIONINSIGHTS_CONNECTION_STRING" = module.applicationinsights.APPLICATIONINSIGHTS_CONNECTION_STRING
+  }
+
+  app_command_line = "pm2 serve /home/site/wwwroot --no-daemon --spa"
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy app service api
+# ------------------------------------------------------------------------------------------------------
+module "appserviceapi" {

```

Comment:
I'd like to name these "web", "api", etc like we hav ein bicep...if possible

Reply:
renamed